spec/injection.spec.ts

```diff
@@ -1152,6 +1164,20 @@ describe('Injector', () => {
             expect((lazyIndividual.value as Child).age).toBe(7);
         });
 
+        it('should resolve an instance of lazy when getLazy invoked with a token', () => {
+            const instance = getInstance();
+
+            instance.register(Child, {
+                tokens: ['child'],
+            });
+
+            const lazy = instance.getLazy<Child>('child');
+
+            expect(lazy instanceof LazyInstance).toBeTruthy();
+            expect(lazy.value.id).toBeDefined();
+            expect(lazy.value.age).toBe(7);
```

Comment:
in these situations I usually test the returned type by trying to do something with the returned object that is only available on that type:

```
lazyChild.callSomeFunction(); // will give a typing error if lazyChild is not correctly typed
```